guardrails/guard.py

```diff
@@ -237,9 +238,10 @@ def __call__(
         msg_history: Optional[List[Dict]] = None,
         metadata: Optional[Dict] = None,
         full_schema_reask: Optional[bool] = None,
+        stream: Optional[bool] = False,
         *args,
         **kwargs,
-    ) -> ValidationOutcome[OT]:
+    ) -> Union[ValidationOutcome[OT], Iterable[str]]:
```

Comment:
This is the best solution we have with the current implementation, but this is still problematic.  Besides unhelpflu intellisense, users using pyright or mypy for typing will have to cast their result and know what to cast it too.

I think the better option would be for streaming to only be available through its own guard method i.e. `guard.stream`.  This way there's no question what the return type would be.  Since the return type of `__call__` is currently dependent on the value of the `stream` kwarg, not just its type, it is not possible to properly overload the method signatures to account for it.

Reply:
logged an issue and we have next steps